File: iosApp/iosApp/Screens/GameInfo/GachaReportSection.swift
import SwiftUI
import UniformTypeIdentifiers
import Shared

// 가챠 효율 리포트 — UIGF/SRGF JSON 가져오기 + 게임별 단가·출현율·풀별·최근5성. (Compose GachaReportSection 대응)
struct GachaReportSection: View {
    @ObservedObject var store: SpendingStore
    let onOpenDashboard: () -> Void
    @Environment(\.glgAccent) private var accent
    @State private var importing = false

    private var stats: GachaStats? { store.gachaStats }
    private var spend: [String: Int64] { store.gachaSpendByGame() }

    var body: some View {
        VStack(alignment: .leading, spacing: 0) {
            HStack {
                HStack(spacing: 6) {
                    Text("가챠 효율 리포트").font(.system(size: 16, weight: .bold))
                    Text("Beta").font(.system(size: 9, weight: .bold)).foregroundStyle(accent.primary)
                        .padding(.horizontal, 6).padding(.vertical, 1)
                        .background(accent.primary.opacity(0.12), in: RoundedRectangle(cornerRadius: 6))
                }
                Spacer()
                if stats != nil {
                    Button { store.clearGachaRecords() } label: {
                        Text("초기화").font(.system(size: 12, weight: .bold)).foregroundStyle(GLGColor.textSecondary)
                    }.buttonStyle(.plain)
                }
            }
            .padding(.bottom, 12)
            if let s = stats { reportContent(s) } else { GLGCard(cornerRadius: 24, padding: 16) { emptyState } }
        }
        .fileImporter(isPresented: $importing, allowedContentTypes: [.json], allowsMultipleSelection: true) { result in
            if case .success(let urls) = result {
                let contents: [String] = urls.compactMap { url in
                    let scoped = url.startAccessingSecurityScopedResource()
                    defer { if scoped { url.stopAccessingSecurityScopedResource() } }
                    return try? String(contentsOf: url, encoding: .utf8)
                }
                if !contents.isEmpty { store.importGachaFromContents(contents) }
            }
        }
    }

    private var emptyState: some View {
        VStack(spacing: 0) {
            ZStack { Circle().fill(accent.primary.opacity(0.12)).frame(width: 52, height: 52)
                Image(systemName: "square.and.arrow.up").font(.system(size: 24)).foregroundStyle(accent.primary) }
            Text("아직 가챠 기록이 없어요").font(.system(size: 14, weight: .bold)).padding(.top, 12)
            Text("UIGF(원신·젠레스) / SRGF·UIGF(스타레일) 표준 JSON을 가져오면\n5성 단가 · 평균 천장 · 획득 히스토리를 분석해 드려요.")
                .font(.system(size: 12)).foregroundStyle(GLGColor.textSecondary).multilineTextAlignment(.center).padding(.top, 6)
            GLGButton(title: "가챠 기록 JSON 가져오기") { importing = true }.padding(.top, 16)
        }
        .frame(maxWidth: .infinity)
    }

    // design_gachareport_mockup.html(B) — 게임별 카드(배지+4통계+운분포 바+최근5성), 첫 카드에 대시보드 진입.
    private let lucky = Color(hex: 0xFF2BB673)
    private let gold = Color(hex: 0xFFE0A93B)
    private let unluckyC = Color(hex: 0xFFE8634A)

    private func reportContent(_ s: GachaStats) -> some View {
        let order = GachaReport.shared.gameOrder
        let games = s.byGame.keys.sorted { (order.firstIndex(of: $0) ?? 99) < (order.firstIndex(of: $1) ?? 99) }
        let poolLabels = GachaReport.shared.poolLabels
        return VStack(alignment: .leading, spacing: 14) {
            ForEach(Array(games.enumerated()), id: \.offset) { idx, gk in
                if let g = s.byGame[gk] {
                    gameCard(gk, g, labels: poolLabels[gk] ?? [:], showDash: idx == 0)
                }
            }
            GLGButton(title: "기록 추가 가져오기") { importing = true }
        }
    }

    private func gameCard(_ gk: String, _ g: GachaGameStat, labels: [String: String], showDash: Bool) -> some View {
        let info = gachaGameInfo(gk)
        let sp = spend[gk] ?? 0
        let cost = (sp > 0 && g.five > 0) ? sp / Int64(g.five) : 0
        let dist = g.luckDist.map { Int(truncating: $0) }
        let distTotal = max(dist.reduce(0, +), 1)
        return GLGCard(cornerRadius: 20, padding: 16) {
            VStack(alignment: .leading, spacing: 0) {
                // 헤더 — 배지 + 게임명
                HStack(spacing: 10) {
                    Text(reportAbbr(gk)).font(.system(size: 11, weight: .heavy)).foregroundStyle(.white)
                        .padding(.horizontal, 7).padding(.vertical, 4)
                        .background(info.color, in: RoundedRectangle(cornerRadius: 7, style: .continuous))
                    Text(info.short).font(.system(size: 15, weight: .bold))
                    Spacer()
                }
                .padding(.bottom, 10)
                // 4 통계
                HStack(spacing: 0) {
                    statCol("\(num(Int(g.total)))", "총 뽑기")
                    statCol("\(num(Int(g.five)))", "5성")
                    statCol(g.avgPity > 0 ? "\(g.avgPity)" : "—", "평균 천장", accent.primary)
                    statCol(cost > 0 ? wonShort(cost) : "—", "5성 단가", accent.primary)
                }
                .padding(.bottom, 12)
                // 운 분포 바
                if Int(g.five) > 0 {
                    HStack {
                        Text("운 분포 (천장 구간)").font(.system(size: 11)).foregroundStyle(GLGColor.textSecondary)
                        Spacer()
                        Text("5성 \(num(Int(g.five)))개").font(.system(size: 11)).foregroundStyle(GLGColor.textSecondary)
                    }
                    .padding(.bottom, 6)
                    GeometryReader { geo in
                        HStack(spacing: 0) {
                            lucky.frame(width: geo.size.width * CGFloat(dist[0]) / CGFloat(distTotal))
                            gold.frame(width: geo.size.width * CGFloat(dist[1]) / CGFloat(distTotal))
                            unluckyC.frame(width: geo.size.width * CGFloat(dist[2]) / CGFloat(distTotal))
                        }
                    }
                    .frame(height: 8).clipShape(Capsule())
                    HStack(spacing: 12) {
                        legendItem(lucky, "~40 행운"); legendItem(gold, "41~74 평균"); legendItem(unluckyC, "75+ 불운")
                    }
                    .padding(.top, 8)
                }
                // 최근 5성
                if !g.recentFive.isEmpty {
                    Text("최근 5성").font(.system(size: 13, weight: .bold)).foregroundStyle(GLGColor.textSecondary).padding(.top, 14).padding(.bottom, 8)
                    FlexibleRow(Array(g.recentFive.enumerated().map { IdxFive(i: $0.offset, name: $0.element.name, pity: Int($0.element.pity)) })) { item in
                        let c: Color = item.pity <= 40 ? lucky : (item.pity >= 75 ? unluckyC : GLGColor.textPrimary)
                        HStack(spacing: 5) {
                            Text(item.name).font(.system(size: 11, weight: .bold)).foregroundStyle(GLGColor.textPrimary)
                            Text("\(item.pity)").font(.system(size: 10, weight: .heavy)).foregroundStyle(c)
                        }
                        .padding(.horizontal, 9).padding(.vertical, 5)
                        .background(Color(hex: 0xFFF3F4F8), in: Capsule())
                    }
                }
                // 대시보드 진입 (첫 카드)
                if showDash {
                    Divider().padding(.top, 13)
                    Button { onOpenDashboard() } label: {
                        HStack {
                            Text("상세 대시보드 (월별·풀별 추이)").font(.system(size: 13, weight: .bold)).foregroundStyle(accent.primary)
                            Spacer()
                            Image(systemName: "chevron.right").font(.system(size: 12, weight: .semibold)).foregroundStyle(accent.primary)
                        }
                        .padding(.vertical, 12)
                    }.buttonStyle(.plain)
                }
            }
        }
    }

    private func statCol(_ value: String, _ label: String, _ color: Color = GLGColor.textPrimary) -> some View {
        VStack(spacing: 2) {
            Text(value).font(.system(size: 20, weight: .bold)).foregroundStyle(color).lineLimit(1).minimumScaleFactor(0.6)
            Text(label).font(.system(size: 10)).foregroundStyle(GLGColor.textSecondary).lineLimit(1)
        }
        .frame(maxWidth: .infinity)
    }
    private func legendItem(_ c: Color, _ text: String) -> some View {
        HStack(spacing: 4) {
            RoundedRectangle(cornerRadius: 2).fill(c).frame(width: 8, height: 8)
            Text(text).font(.system(size: 10)).foregroundStyle(GLGColor.textSecondary)
        }
    }
    private func reportAbbr(_ gk: String) -> String {
        switch gk { case "genshin": return "GI"; case "hsr", "starrail": return "HSR"; case "zzz": return "ZZZ"; default: return gk.uppercased() }
    }
    private func wonShort(_ v: Int64) -> String {
        v >= 10000 ? "\(fixed(Double(v) / 10000, 1))만" : won(v)
    }
}

private struct IdxFive: Hashable { let i: Int; let name: String; let pity: Int }

private struct IdxStr: Hashable { let i: Int; let s: String }

// 프로필 쇼케이스 — Enka.Network UID 조회. (Compose ProfileShowcaseSection 대응)
struct ProfileShowcaseSection: View {
    @ObservedObject var store: SpendingStore
    @Environment(\.glgAccent) private var accent
    @State private var game = "genshin"
    @State private var uid = ""
    @State private var showHoyolab = false
    @State private var loadTask: Task<Void, Never>? = nil

    private let gold = Color(hex: 0xFFD8A12E)
    private let purple = Color(hex: 0xFF9B6BD6)
    private let maxRetries = 2

    // 저장된 enka 조회 UID(과거 조회 시 KMP가 자동 저장) — 1순위.
    private func savedUid(_ g: String) -> String { g == "genshin" ? store.enkaGiUid : store.enkaHsrUid }
    // HoYoLAB 연동으로 가져온 게임 UID — 2순위(저장된 enka UID가 없을 때).
    private func hoyoUid(_ g: String) -> String { g == "genshin" ? store.hoyolabConfig.genshinUid : store.hoyolabConfig.hsrUid }
    private func defaultUid(_ g: String) -> String { let s = savedUid(g); return s.isEmpty ? hoyoUid(g) : s }

    // UID가 있으면 자동 조회. (clearEnkaResult는 비동기 반영이라 enkaResult==nil 가드를 쓰면
    //  탭 전환 시 직전 게임 결과가 아직 남아 조회가 스킵되는 레이스가 생긴다 → 무조건 조회)
    private func autoLoad(_ g: String) {
        let d = defaultUid(g)
        uid = d
        if !d.isEmpty { load(game: g, uid: d) }
    }

    // 타임아웃/네트워크 오류는 일시적이므로 자동 새로고침(재시도) 대상.
    private func isRetriable(_ err: String) -> Bool { err.contains("네트워크") || err.contains("요청이 많") }

    /// Enka 조회 + 타임아웃 워치독 — 조회가 타임아웃/네트워크 오류로 끝나면 자동 재시도(최대 maxRetries).
    private func load(game g: String, uid u: String, attempt: Int = 0) {
        store.loadEnkaProfile(game: g, uid: u)
        loadTask?.cancel()
        loadTask = Task {
            // 조회가 끝날 때까지(로딩 false) 대기 — Net 타임아웃(12s) + 여유.
            for _ in 0..<80 {
                try? await Task.sleep(nanoseconds: 200_000_000)
                if Task.isCancelled { return }
                if !store.enkaLoading { break }
            }
            if Task.isCancelled { return }
            // 타임아웃/네트워크 오류 + 재시도 여유가 있으면 잠시 후 자동 새로고침.
            if let err = store.enkaResult?.error, isRetriable(err), attempt < maxRetries {
                try? await Task.sleep(nanoseconds: 1_200_000_000)
                if Task.isCancelled { return }
                load(game: g, uid: u, attempt: attempt + 1)
            }
        }
    }

    var body: some View {
        let hasUid = !defaultUid(game).isEmpty || !uid.isEmpty
        return VStack(alignment: .leading, spacing: 0) {
            Text("프로필 쇼케이스").font(.system(size: 16, weight: .bold)).padding(.bottom, 12)
            GLGCard(cornerRadius: 24, padding: 16) {
                VStack(alignment: .leading, spacing: 0) {
                    HStack(spacing: 8) {
                        gameTab("원신", game == "genshin", Color(hex: 0xFF4F8EF7)) { switchGame("genshin") }
                        gameTab("스타레일", game == "hsr", Color(hex: 0xFFB06BFF)) { switchGame("hsr") }
                    }
                    // UID가 없으면 HoYoLAB 연동 진입을 노출(연동하면 UID 자동 확보).
                    if !hasUid {
                        hoyolabPrompt.padding(.top, 14)
                    }
                    HStack(spacing: 8) {
                        TextField("UID 입력 (예: 800000000)", text: $uid).keyboardType(.numberPad).textFieldStyle(.plain).glgPillField()
                            .onChange(of: uid) { uid = $0.filter(\.isNumber) }
                        Button("조회") { load(game: game, uid: uid) }
                            .buttonStyle(.borderedProminent)
                            .tint(accent.primary)
                            .controlSize(.large)
                            .disabled(uid.isEmpty)
                    }
                    .padding(.top, 12)
                    Text("게임 내 '프로필 표시(쇼케이스)'에 올린 캐릭터만 조회돼요. 조회한 UID는 이 기기에 저장돼 다음엔 자동으로 불러와요.")
                        .font(.system(size: 11)).foregroundStyle(Color(.systemGray3)).padding(.top, 8)
                    resultView
                }
            }
        }
        .navigationDestination(isPresented: $showHoyolab) { HoyolabLinkView(store: store) { showHoyolab = false } }
        // 진입 시 저장된/연동 UID가 있으면 자동 조회. (연동 페이지에서 돌아올 때도 재실행)
        .onAppear { autoLoad(game) }
        .onDisappear { loadTask?.cancel() }
    }

    private func switchGame(_ g: String) {
        game = g
        store.clearEnkaResult()
        autoLoad(g)
    }

    // ── UID 없음: HoYoLAB 연동 진입 CTA ──
    private var hoyolabPrompt: some View {
        Button { showHoyolab = true } label: {
            HStack(spacing: 10) {
                ZStack {
                    RoundedRectangle(cornerRadius: 10, style: .continuous).fill(accent.primary.opacity(0.12)).frame(width: 38, height: 38)
                    Image(systemName: "link").font(.system(size: 16, weight: .semibold)).foregroundStyle(accent.primary)
                }
                VStack(alignment: .leading, spacing: 2) {
                    Text("HoYoLAB 연동하고 UID 자동 가져오기").font(.system(size: 13, weight: .bold)).foregroundStyle(GLGColor.textPrimary)
                    Text("연동하면 UID 입력 없이 바로 조회돼요").font(.system(size: 11)).foregroundStyle(GLGColor.textSecondary)
                }
                Spacer(minLength: 6)
                Image(systemName: "chevron.right").font(.system(size: 13, weight: .semibold)).foregroundStyle(GLGColor.textSecondary)
            }
            .padding(12)
            .background(accent.primary.opacity(0.06), in: RoundedRectangle(cornerRadius: 14, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 14, style: .continuous).stroke(accent.primary.opacity(0.18), lineWidth: 1))
        }.buttonStyle(.plain)
    }

    @ViewBuilder private var resultView: some View {
        if store.enkaLoading {
            ProgressView().tint(accent.primary).frame(maxWidth: .infinity).padding(.vertical, 16)
        } else if let err = store.enkaResult?.error {
            Text(err).font(.system(size: 13)).foregroundStyle(Color(hex: 0xFFDC2626)).padding(.top, 14)
        } else if let p = store.enkaResult?.profile {
            VStack(alignment: .leading, spacing: 0) {
                Divider().padding(.vertical, 14)
                Text(p.nickname.isEmpty ? "이름 없음" : p.nickname).font(.system(size: 17, weight: .bold))
                Text("Lv.\(p.level) · 월드 레벨 \(p.worldLevel)").font(.system(size: 12)).foregroundStyle(GLGColor.textSecondary).padding(.top, 2)
                if !p.signature.isEmpty { Text(p.signature).font(.system(size: 12)).foregroundStyle(GLGColor.textSecondary).padding(.top, 4) }
                if p.chars.isEmpty {
                    Text("쇼케이스에 등록된 캐릭터가 없어요").font(.system(size: 12)).foregroundStyle(GLGColor.textSecondary).padding(.top, 14)
                } else {
                    let cols = [GridItem(.flexible()), GridItem(.flexible()), GridItem(.flexible())]
                    LazyVGrid(columns: cols, spacing: 8) {
                        ForEach(Array(p.chars.enumerated()), id: \.offset) { _, c in charCard(c) }
                    }
                    .padding(.top, 14)
                }
            }
        }
    }

    private func charCard(_ c: EnkaChar) -> some View {
        let color = c.rarity >= 5 ? gold : purple
        let rankLabel: String? = game == "genshin"
            ? (c.rank < 0 ? nil : (c.rank == 0 ? "명함" : "\(c.rank)돌"))
            : (c.rank > 0 ? "\(c.rank)성혼" : nil)
        return VStack(spacing: 6) {
            ZStack {
                RoundedRectangle(cornerRadius: 12).fill(color.opacity(0.18)).aspectRatio(1, contentMode: .fit)
                if let icon = c.iconUrl, let u = URL(string: icon) {
                    AsyncImage(url: u) { $0.resizable().scaledToFill() } placeholder: { Color.clear }
                        .clipShape(RoundedRectangle(cornerRadius: 12))
                } else {
                    Text(String(c.name.prefix(1))).font(.system(size: 22, weight: .bold)).foregroundStyle(color)
                }
                VStack {
                    HStack {
                        Spacer()
                        if let r = rankLabel {
                            Text(r).font(.system(size: 9, weight: .bold)).foregroundStyle(.white)
                                .padding(.horizontal, 5).padding(.vertical, 1).background(color)
                                .clipShape(RoundedCorner(radius: 8, corners: [.bottomLeft]))
                        }
                    }
                    Spacer()
                    HStack {
                        Text("Lv.\(c.level)").font(.system(size: 9, weight: .bold)).foregroundStyle(.white)
                            .padding(.horizontal, 5).padding(.vertical, 1).background(Color.black.opacity(0.8))
                            .clipShape(RoundedCorner(radius: 8, corners: [.topRight]))
                        Spacer()
                    }
                }
            }
            Text(c.name).font(.system(size: 12, weight: .bold)).lineLimit(1)
            HStack(spacing: 0) {
                Text(String(repeating: "★", count: min(max(Int(c.rarity), 1), 5))).font(.system(size: 9)).foregroundStyle(color)
                if !c.element.isEmpty { Text(" · \(c.element)").font(.system(size: 10)).foregroundStyle(GLGColor.textSecondary).lineLimit(1) }
            }
        }
        .padding(8)
        .background(color.opacity(0.10), in: RoundedRectangle(cornerRadius: 14))
    }

    private func gameTab(_ label: String, _ selected: Bool, _ color: Color, _ action: @escaping () -> Void) -> some View {
        Button(action: action) {
            Text(label).font(.system(size: 12, weight: .bold)).foregroundStyle(selected ? .white : GLGColor.textSecondary)
                .padding(.horizontal, 14).padding(.vertical, 6)
                .background(selected ? color : Color(hex: 0xFFF2F2F6), in: Capsule())
        }.buttonStyle(.plain)
    }
}

/// 특정 모서리만 둥글게 (배지용).
struct RoundedCorner: Shape {
    var radius: CGFloat = 8
    var corners: UIRectCorner = .allCorners
    func path(in rect: CGRect) -> Path {
        Path(UIBezierPath(roundedRect: rect, byRoundingCorners: corners, cornerRadii: CGSize(width: radius, height: radius)).cgPath)
    }
}

File: GL_Android/src/main/java/com/gatcha/log/ui/game/GachaReportSection.kt
package com.gatcha.log.ui.game

import android.net.Uri
import androidx.activity.compose.rememberLauncherForActivityResult
import androidx.activity.result.contract.ActivityResultContracts
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material.icons.filled.FileUpload
import com.gatcha.log.data.GachaGameStat
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalUriHandler
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.gatcha.log.data.GachaReport
import com.gatcha.log.data.GachaStats
import com.gatcha.log.ui.components.GlassCard
import com.gatcha.log.ui.components.GlgButton
import com.gatcha.log.ui.components.GlgOutlineButton
import com.gatcha.log.ui.theme.DividerColor
import com.gatcha.log.ui.theme.toColor
import com.gatcha.log.ui.theme.LocalAccent
import com.gatcha.log.ui.theme.TextPrimary
import com.gatcha.log.ui.theme.TextSecondary
import com.gatcha.log.util.num
import com.gatcha.log.util.won

@Composable
fun GachaReportSection(
    stats: GachaStats?,
    spendByGameKey: Map<String, Long>,
    onImport: (List<Uri>) -> Unit,
    onClear: () -> Unit,
    onOpenDashboard: () -> Unit = {},
) {
    val accent = LocalAccent.current
    val picker = rememberLauncherForActivityResult(ActivityResultContracts.OpenMultipleDocuments()) { uris ->
        if (uris.isNotEmpty()) onImport(uris)
    }
    // MIME 필터를 좁히면 일부 제공자(삼성 SAF·클라우드)가 JSON 을 회색 처리해 선택 불가 → */* 로 전부 허용
    val openPicker = { picker.launch(arrayOf("*/*")) }

    Row(
        modifier = Modifier.fillMaxWidth().padding(bottom = 12.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Row(verticalAlignment = Alignment.CenterVertically) {
            Text("가챠 효율 리포트", fontSize = 16.sp, fontWeight = FontWeight.Bold)
            Spacer(Modifier.width(6.dp))
            Surface(color = accent.copy(alpha = 0.12f), shape = RoundedCornerShape(6.dp)) {
                Text("Beta", fontSize = 9.sp, fontWeight = FontWeight.Bold, color = accent, modifier = Modifier.padding(horizontal = 6.dp, vertical = 1.dp))
            }
        }
        if (stats != null) {
            Text("초기화", fontSize = 12.sp, color = TextSecondary, fontWeight = FontWeight.Bold, modifier = Modifier.clickable { onClear() }.padding(4.dp))
        }
    }

    if (stats == null) {
        GlassCard(shape = RoundedCornerShape(24.dp), modifier = Modifier.fillMaxWidth()) {
            Column(Modifier.padding(16.dp)) { EmptyState(onImport = openPicker) }
        }
    } else {
        ReportContent(stats, spendByGameKey, onImport = openPicker, onOpenDashboard = onOpenDashboard)
    }
}

// 운 분포색
private val Lucky = Color(0xFF2BB673)
private val Avg = Color(0xFFE0A93B)
private val Unlucky = Color(0xFFE8634A)
private fun reportAbbr(gk: String) = when (gk) { "genshin" -> "GI"; "hsr", "starrail" -> "HSR"; "zzz" -> "ZZZ"; else -> gk.uppercase() }
private fun wonShort(v: Long): String = if (v >= 10000) "%.1f만".format(v / 10000.0) else won(v)

@Composable
private fun EmptyState(onImport: () -> Unit) {
    val uriHandler = LocalUriHandler.current
    Column(Modifier.fillMaxWidth().padding(vertical = 8.dp), horizontalAlignment = Alignment.CenterHorizontally) {
        Box(Modifier.size(52.dp).clip(CircleShape).background(LocalAccent.current.copy(alpha = 0.12f)), contentAlignment = Alignment.Center) {
            Icon(Icons.Default.FileUpload, null, tint = LocalAccent.current, modifier = Modifier.size(26.dp))
        }
        Spacer(Modifier.height(12.dp))
        Text("아직 가챠 기록이 없어요", fontSize = 14.sp, fontWeight = FontWeight.Bold)
        Spacer(Modifier.height(6.dp))
        Text(
            "UIGF(원신·젠레스) / SRGF·UIGF(스타레일) 표준 JSON을 가져오면\n5성 단가 · 평균 천장 · 획득 히스토리를 분석해 드려요.",
            fontSize = 12.sp, color = TextSecondary,
            textAlign = androidx.compose.ui.text.style.TextAlign.Center, lineHeight = 17.sp,
        )
        Spacer(Modifier.height(16.dp))
        GlgButton("가챠 기록 JSON 가져오기", onClick = onImport, modifier = Modifier.fillMaxWidth())
        Spacer(Modifier.height(10.dp))
        Text(
            "UIGF/SRGF가 뭔가요?",
            fontSize = 12.sp, color = LocalAccent.current, fontWeight = FontWeight.Medium,
            modifier = Modifier.clickable { uriHandler.openUri("https://uigf.org/") },
        )
    }
}

@Composable
private fun ReportContent(stats: GachaStats, spendByGameKey: Map<String, Long>, onImport: () -> Unit, onOpenDashboard: () -> Unit) {
    val games = stats.byGame.keys.sortedBy { GachaReport.gameOrder.indexOf(it).let { i -> if (i < 0) 99 else i } }
    Column(verticalArrangement = Arrangement.spacedBy(14.dp)) {
        games.forEachIndexed { idx, gk ->
            val g = stats.byGame[gk] ?: return@forEachIndexed
            GameCard(gk, g, spendByGameKey[gk] ?: 0L, showDash = idx == 0, onOpenDashboard)
        }
        GlgButton("기록 추가 가져오기", onClick = onImport, modifier = Modifier.fillMaxWidth(), height = 46.dp)
    }
}

// design_gachareport_mockup.html(B) — 게임 카드: 배지+4통계+운분포 바+최근5성, 첫 카드에 대시보드 진입.
@OptIn(ExperimentalLayoutApi::class)
@Composable
private fun GameCard(gk: String, g: GachaGameStat, spend: Long, showDash: Boolean, onOpenDashboard: () -> Unit) {
    val accent = LocalAccent.current
    val (shortName, _, color) = GachaReport.gameInfo[gk] ?: Triple(gk, gk, 0xFF888888L)
    val cost = if (spend > 0 && g.five > 0) spend / g.five else 0L
    val dist = g.luckDist
    GlassCard(shape = RoundedCornerShape(20.dp), modifier = Modifier.fillMaxWidth()) {
        Column(Modifier.padding(16.dp)) {
            // 헤더 — 배지 + 게임명
            Row(verticalAlignment = Alignment.CenterVertically) {
                Surface(color = color.toColor(), shape = RoundedCornerShape(7.dp)) {
                    Text(reportAbbr(gk), fontSize = 11.sp, fontWeight = FontWeight.Black, color = Color.White, modifier = Modifier.padding(horizontal = 7.dp, vertical = 4.dp))
                }
                Spacer(Modifier.width(10.dp))
                Text(shortName, fontSize = 15.sp, fontWeight = FontWeight.Bold, color = TextPrimary)
            }
            Spacer(Modifier.height(10.dp))
            // 4 통계
            Row(Modifier.fillMaxWidth()) {
                StatCol(num(g.total), "총 뽑기", Modifier.weight(1f))
                StatCol(num(g.five), "5성", Modifier.weight(1f))
                StatCol(if (g.avgPity > 0) "${g.avgPity}" else "—", "평균 천장", Modifier.weight(1f), accent)
                StatCol(if (cost > 0) wonShort(cost) else "—", "5성 단가", Modifier.weight(1f), accent)
            }
            Spacer(Modifier.height(12.dp))
            // 운 분포 바
            if (g.five > 0) {
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Text("운 분포 (천장 구간)", fontSize = 11.sp, color = TextSecondary)
                    Text("5성 ${num(g.five)}개", fontSize = 11.sp, color = TextSecondary)
                }
                Spacer(Modifier.height(6.dp))
                Row(Modifier.fillMaxWidth().height(8.dp).clip(CircleShape)) {
                    if (dist[0] > 0) Box(Modifier.weight(dist[0].toFloat()).fillMaxHeight().background(Lucky))
                    if (dist[1] > 0) Box(Modifier.weight(dist[1].toFloat()).fillMaxHeight().background(Avg))
                    if (dist[2] > 0) Box(Modifier.weight(dist[2].toFloat()).fillMaxHeight().background(Unlucky))
                }
                Spacer(Modifier.height(8.dp))
                Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
                    Legend(Lucky, "~40 행운"); Legend(Avg, "41~74 평균"); Legend(Unlucky, "75+ 불운")
                }
            }
            // 최근 5성
            if (g.recentFive.isNotEmpty()) {
                Spacer(Modifier.height(14.dp))
                Text("최근 5성", fontSize = 13.sp, color = TextSecondary, fontWeight = FontWeight.Bold)
                Spacer(Modifier.height(8.dp))
                FlowRow(horizontalArrangement = Arrangement.spacedBy(6.dp), verticalArrangement = Arrangement.spacedBy(6.dp)) {
                    g.recentFive.forEach { f -> ReportChip(f.name, f.pity) }
                }
            }
            // 대시보드 진입 (첫 카드)
            if (showDash) {
                Spacer(Modifier.height(13.dp)); HorizontalDivider(color = DividerColor)
                Row(
                    Modifier.fillMaxWidth().clickable { onOpenDashboard() }.padding(vertical = 12.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text("상세 대시보드 (월별·풀별 추이)", fontSize = 13.sp, fontWeight = FontWeight.Bold, color = accent)
                    Spacer(Modifier.weight(1f))
                    Icon(Icons.Default.ChevronRight, null, tint = accent, modifier = Modifier.size(16.dp))
                }
            }
        }
    }
}

@Composable
private fun StatCol(value: String, label: String, modifier: Modifier = Modifier, color: Color = TextPrimary) {
    Column(modifier, horizontalAlignment = Alignment.CenterHorizontally) {
        Text(value, fontSize = 20.sp, fontWeight = FontWeight.Bold, color = color, maxLines = 1)
        Spacer(Modifier.height(2.dp))
        Text(label, fontSize = 10.sp, color = TextSecondary)
    }
}

@Composable
private fun Legend(c: Color, text: String) {
    Row(verticalAlignment = Alignment.CenterVertically) {
        Box(Modifier.size(8.dp).clip(RoundedCornerShape(2.dp)).background(c))
        Spacer(Modifier.width(4.dp))
        Text(text, fontSize = 10.sp, color = TextSecondary)
    }
}

@Composable
private fun ReportChip(name: String, pity: Int) {
    val c = if (pity <= 40) Lucky else if (pity >= 75) Unlucky else TextPrimary
    Surface(color = Color(0xFFF3F4F8), shape = CircleShape) {
        Row(Modifier.padding(horizontal = 9.dp, vertical = 5.dp), verticalAlignment = Alignment.CenterVertically) {
            Text(name, fontSize = 11.sp, fontWeight = FontWeight.Bold, color = TextPrimary)
            Spacer(Modifier.width(5.dp))
            Text("$pity", fontSize = 10.sp, fontWeight = FontWeight.Black, color = c)
        }
    }
}
